flow-data/src/main/java/com/vaadin/flow/data/binder/Binder.java

```diff
@@ -282,6 +282,18 @@ default BindingValidationStatus<TARGET> validate() {
          * @return A boolean value
          */
         boolean isConvertBackToPresentation();
+
+        /**
+         * Checks whether the related Binder instance has marked the field as
+         * having uncommitted changes.
+         *
+         * @throws IllegalStateException
+         *             if {@code binder} is null. Ensure that the binder has
+         *             been declared here.
+         *
+         * @return True if field has uncommitted changes
+         */
+        boolean hasChanges() throws IllegalStateException;
```

Comment:
I would skip the exception declaration, as it is a runtime exception.
I think that mentioning the exception in the javadoc is enough.

```suggestion
        boolean hasChanges() throws IllegalStateException;
```

Reply:
Fair enough